design-documents/service-isolation/split-extensions.md

```diff
@@ -61,20 +61,41 @@ Cons:
 
 ### New module - old composer
 
-This strategy proposes to expose all new modules via one (currently existing) `composer.json` and more described in [this proposal](https://github.com/magento/architecture/issues/88).
+This strategy proposes to expose all new modules via one (currently existing) `composer.json` and came from [this proposal](https://github.com/magento/architecture/issues/88).
 The main idea that extensions will use the same dependency in `composer.json` which will be distributed as metapackage. Magento core modules will depend on new modules like `catalog-ui` but not to `catalog` metapackage, such metapackage will be distributed separately.
+Such strategy assumes that existing modules will use the old namespaces but new modules will have the new namespaces.
 
  ```json
 "require": {
     "magento/module-catalog": "^103.0.0|^104.0.0"
 }
 ```
 
+Unlike the [original proposal]((https://github.com/magento/architecture/issues/88)), the `composer.json` probably will have another structure. As an example for `catalog` metapackage:
+
+```json
+"require": {
+    "magento/module-catalog-api": "^104.0.0",
+    "magento/module-catalog-adminui": "^104.0.0",
+    "magento/module-catalog-ui": "^104.0.0"
+}
+```
+
+The original proposal uses `self.version` for `catalog` related modules but such approach does not allow bump modules versions independently and update just their versions in the metapackage (according to the original proposal, all versions of `catalog` related modules should be bumped).
+Also, the usage of `provide` package link, probably is not the best option, as in our case `catalog` metapackage knows about exact packages not virtual interfaces.
+
 Pros:
  - No changes in the dependant code
  - Allows to specify multiple package versions
   
 Cons:
+ - Introduce a metapackage with all needed dependencies
  - `composer.json` anyway should be changed to specify needed version of module
- - All modules and extensions still have coupling to all dependant code even if they do not use it
- - Such approach won't get benefits of modules decoupling as an extension still depends on the whole component
\ No newline at end of file
+ - All extensions still have coupling to all dependant code even if they do not use it
+ - Such approach won't get benefits of modules decoupling as an extension still depends on the whole component
```

Comment:
To avoid confusion, could you clarify that this is only for extensions that are not decomposed? Maybe combine this and last one?

Reply:
Updated.